screenpipe-audio/src/stt.rs

```diff
@@ -737,70 +738,35 @@ pub async fn create_whisper_channel(
         UnboundedSender<TranscriptionResult>,
         UnboundedReceiver<TranscriptionResult>,
     ) = unbounded_channel();
-
-    let shutdown_flag = Arc::new(AtomicBool::new(false));
-    let shutdown_flag_clone = shutdown_flag.clone();
-
+    let mut vad_engine: Box<dyn VadEngine> = match vad_engine {
+        VadEngineEnum::WebRtc => Box::new(WebRtcVad::new()),
+        VadEngineEnum::Silero => Box::new(SileroVad::new().unwrap()),
+    };
     tokio::spawn(async move {
         loop {
-            if shutdown_flag_clone.load(Ordering::Relaxed) {
-                info!("Whisper channel shutting down");
-                break;
-            }
-
             tokio::select! {
                 Some(input) = input_receiver.recv() => {
                     let timestamp = SystemTime::now()
                         .duration_since(UNIX_EPOCH)
                         .expect("Time went backwards")
                         .as_secs();
 
-                    let transcription_result = if cfg!(target_os = "macos") {
-                        #[cfg(target_os = "macos")]
-                        {
-                            autoreleasepool(|| {
```

Comment:
@itsdheerajdp autoreleasepool is necessary https://github.com/huggingface/candle/issues/2271